modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue

```diff
@@ -399,6 +400,47 @@ export default {
     }
   },
   methods: {
+    updateFieldState(fieldState) {
+      this.tabKey = cuid();
+      for (const key in this.groups) {
+        this.groups[key].fields.forEach(field => {
+          if (fieldState[field]) {
+            this.fieldErrors[key][field] = fieldState[field].error;
+          }
+        });
+      }
+      this.updateErrorCount();
+    },
+    updateErrorCount() {
+      let count = 0;
+      for (const key in this.fieldErrors) {
+        for (const tabKey in this.fieldErrors[key]) {
+          if (this.fieldErrors[key][tabKey]) {
+            count++;
+          }
+        }
+      }
+      this.errorCount = count;
+    },
+    focusNextError() {
+      let field;
+      for (const key in this.fieldErrors) {
+        for (const tabKey in this.fieldErrors[key]) {
+          if (this.fieldErrors[key][tabKey] && !field) {
+            field = this.schema.filter(item => {
+              return item.name === tabKey;
+            })[0];
+            this.switchPane(field.group.name);
+            this.getAposSchema(field).scrollFieldIntoView(field.name);
+          }
+        }
+      }
+    },
+    prepErrors() {
```

Comment:
wish this could somehow be more automatic but states are aggregated as schemas send up their field's states. Would take feedback on another approach